### PR TITLE
ImplicitRole doesn't have wildcard labels

### LINKED ISSUE #7618: `GetAppServers` and `GetDatabaseServers` APIs no longer respect RBAC
### Description

**What happened**:

`tsh db ls` and `tsh app ls` commands (as well as respective tabs in the UI) no longer respect `db_labels` and `app_labels` in the user role and show all entries.

1. Create a role with app/db labels e.g. 
```yaml
spec:
  allow:
    app_labels:
      env: prod
    db_labels:
      env: aws
```
2. Registed a few apps/dbs with different labels (apps/dbs don't have to be actually working, just as long as there are registered).
3. Login with `tsh login` and use `tsh app ls`/`tsh db ls` to see that it always displays everything. Same for UI.

Previously, those calls used auth_with_roles.go methods which would filter out objects user doesn't have access to:

[GetAppServers](https://github.com/gravitational/teleport/blob/v7.0.0-beta.2/lib/auth/auth_with_roles.go#L2798-L2811)
[GetDatabaseServers](https://github.com/gravitational/teleport/blob/v7.0.0-beta.2/lib/auth/auth_with_roles.go#L2682-L2689)

Did this change?

**What you expected to happen**:

Those commands/tabs should only display apps/databases user has access to.

### Server Details
- Teleport version (run `teleport version`): 7.0.0-beta.2

### Client Details
- Tsh version (`tsh version`): 7.0.0-beta.1

### PR DESCRIPTION
**Pull request by Joerger**
Change `NewImplicitRole` to return a constant role value so that wildcard labels aren't set. Implicit role is implicitly used as a role for `GetAppServers` and `GetDatabaseServers`, so the wildcard labels caused those two functions to return all servers despite the users's actual role having no labels.

In the future, we should instead make `V4` the default role version so that wildcard labels aren't set unless explicitly listed - https://github.com/gravitational/teleport/pull/7632

Closes #7618

### CODE REVIEW

**Review by r0mant — COMMENTED** [review]
@Joerger This fix looks much less scary than the other one but before merging let's please write a test to make sure this doesn't regress again.

File: lib/auth/auth_with_roles_test.go [hunk 1]
```diff
@@ -355,3 +355,152 @@ func serverWithAllowRules(t *testing.T, srv *TestAuthServer, allowRules []types.
 		context:    *authContext,
 	}
 }
+
+func TestGetDatabaseServers(t *testing.T) {
```
Comment (tcsc) [inline lib/auth/auth_with_roles_test.go:359]: How do these tests relate to changing the way implicit roles are created? (I have no complaints if they're _not_ related - but maybe add a note in the PR description)

**Review by tcsc — APPROVED** [review]

File: lib/auth/auth_with_roles_test.go (hunk 1 above)
Comment (Joerger) [reply lib/auth/auth_with_roles_test.go:359]: Updated the description, those tests would fail if the implicit role change isn't made.

**Review by fspmarshall — APPROVED** [review]

**Review by andrejtokarcik — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
@Joerger Somehow I didn't notice that you already added a test here. Sorry for delay!

**Pull request merged**